### PR TITLE
Addding the AzureML example

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes
Added an example with a README to run pipelines using the AzureML step operator.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by strickvl — APPROVED** [review]

File: examples/azureml_step_operator/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+executions.
+* The **local orchestrator** which will be executing your pipelines steps.
+* The **azure artifact store** which will be responsible for storing the
+artifacts of your pipeline.
+* The **azureml step operator** which will be utilized to run the training step 
+on Azure.
+
+```bash
+zenml artifact-store register azure_store \
+    --path <AZURE_BLOB_CONTAINER_PATH>
```
Comment (wjayesh) [inline examples/azureml_step_operator/README.md:69]:
```suggestion
    --type azure
    --path <AZURE_BLOB_CONTAINER_PATH>
```
small fix to the artifact store registration

File: examples/azureml_step_operator/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+zenml step-operator register azureml \
+    --subscription_id <AZURE_SUBSCRIPTION_ID> \
```
Comment (wjayesh) [inline examples/azureml_step_operator/README.md:72]:
```suggestion
        --type azureml \
        --subscription_id <AZURE_SUBSCRIPTION_ID> \
```

File: examples/azureml_step_operator/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    --resource_group <AZURE_RESOURCE_GROUP> \
+    --workspace_name <AZURE_WORKSPACE_NAME> \
```
Comment (schustmi) [inline examples/azureml_step_operator/README.md:74]: Mentioned something similar on @wjayesh's PR, but I think registering stack components requires arguments of the form --key=value, the `=` being required so the parsing works correctly. This is due to the fact that those arguments are different per flavor of the stack component (e.g. sagemaker step operator has different properties than the azureml one) and the parsing isn't handled by click but by some custom code. If this changed recently and it actually works like this, feel free to ignore

**Review by schustmi — CHANGES_REQUESTED** [review]
LGTM, just mentioned one potential issue

File: examples/azureml_step_operator/README.md (hunk 1 above)
Comment (bcdurak) [reply examples/azureml_step_operator/README.md]: nice catch! fixed it!

File: examples/azureml_step_operator/README.md (hunk 2 above)
Comment (bcdurak) [reply examples/azureml_step_operator/README.md]: nice catch! fixed it!

File: examples/azureml_step_operator/README.md (hunk 3 above)
Comment (bcdurak) [reply examples/azureml_step_operator/README.md]: You are absolutely right! Fixed it!

**Review by schustmi — APPROVED** [review]

**Pull request merged**